src/OpenApi/src/Services/Schemas/OpenApiSchemaStore.cs

```diff
@@ -14,6 +14,8 @@ namespace Microsoft.AspNetCore.OpenApi;
 /// </summary>
 internal sealed class OpenApiSchemaStore
 {
+    // Matches default MaxValidationDepth and MaxModelBindingRecursionDepth used by MVC.
+    private readonly int MaxSchemaReferenceRecursionDepth = 32;
```

Comment:
`MaxValidationDepth` and `MaxModelBindingRecursionDepth` are both configurable via `MvcOptions`. Do we think this will need to be configurable too?

Reply:
In a future release, yes, but we will have to live with it being un-configurable for .NET 9 for now. :/

Reply:
Per off-line discussion, we're relying on STJ's recursion depth check which do have a configurable option via `JsonSerializerOptions.MaxDepth`.